docs/proposals/20220105-observatorium-alerting.md

```diff
@@ -0,0 +1,124 @@
+
+## Why
+
+Alerting is a critical component of reliable monitoring. We can build automation or notify humans in a reactive or proactive manner instead of watching dashboards.
+
+### Pitfalls of the current solution
+
+Observatorium did not have alerting defined before.
+
+## Goals
+
+Definition: 
+* Alert Routing: Configuration of Alertmanager related to receiver target config and its routing. 
+
+Goals:
+* Tenant should be able to use alerting rules.
+  * Alerting Rules via GET and PUT via HTTP (Rules API)
+* Tenants should be able to read what is the status of configured alerts (Prometheus Alert API GET)
+* Tenants are able to route alerts:
+  * Alerting Routing via GET and PUT via HTTP API.
+* Both alerts and alerting rules should be multi-tenant.
+
+## Non-Goals
+
+* (?) UI for Alerts/Rules, Alert Status, AM Routing, AM Alerts …
+* (?) Alertmanager Alerts API, Config/Status
+* Silences
+* Inhibits
+* Scaling Rulers
+* Scaling Alertmanager beyond 3 replicas.
+* Routing tester.
+
+## Audience of this proposal
+
+Observatorium Devs and Users.
+
+## How
+
+Within work for [Rules API proposal](https://observatorium.io/docs/proposals/20201019-prometheus-rules-in-observatorium-api.md/) we already added Rules APIs for reading and configuring rules which include both recording and alerting rules.
+
+For example, to save alerts we can already PUT the following object to `/api/metrics/v1/{tenant}/api/v1/rules/raw`
+
+```
+groups:
+  - name: test-oidc
+    interval: 5s
+    rules:
+      - alert: HighRequestLatency
+        expr: job:request_latency_seconds:mean5m{job="second"} > 0.5
+        for: 10m
+        labels:
+          team: alpha
+```
+
+Such rule will be saved through Observatorium API, then Rule-Store to object storage for the specified tenant. It can be then listed through a similar endpoint and GET.
+
+### Alerts
+
+Such configuration will be then eventually synchronized with tenant’s Rulers using ConfigSync. The Ruler itself has to communicate on 4 levels. All connections are presented in the diagram below.
+
+![ruler-connections-diagram](../assets/alerting-ruler.png)
+
+1. Alerts will be evaluated based on periodic requests to any replica of  HTTP Query API of Query-frontend (Question: Or querier?). 
+2. Triggered alerts will be pushed to all replicas of  Alertmanager (discussed later) through AM HTTP API.
+3. It exposes gRPC Rules API which is consumed by Queriers.
+4. It saves samples to any replica of Receiver Router (or RouterIngestor). 
+
+### Alert eval status
+
+We expect Rulers to be connected to Queriers through gRPC Rules API (NOTE, not Store API as we have now). This allows queries to be able to present federate views through API. In order to fulfil the full story we also need:
```

Comment:
```suggestion
We expect Rulers to be connected to Queriers through gRPC Rules API (NOTE, not Store API as we have now). This allows queries to be able to present federated views through the API. In order to fulfil the full story we also need:
```

Reply:
> This allows queries to be able to present federated views through the API

Observatorium, rules or querier HTTP alerts API?

Reply:
We are talking about Thanos [gRPC Rules API](https://github.com/thanos-io/thanos/blob/main/pkg/rules/rulespb/rpc.proto), which is essentially same as [Prometheus Rules HTTP API](https://prometheus.io/docs/prometheus/latest/querying/api/#rules).

We need to add a TODO because we are missing the [Prometheus Alerts API](https://prometheus.io/docs/prometheus/latest/querying/api/#alerts) in Thanos.

Reply:
I actually have idea how to avoid this extra work, added explanation!